Add troubleshooting entry for recurring HTTP 407 issue with SAP Cloud Connector (#220)
SAP/cloud-sdk: docs/java/troubleshooting.mdx

  Please check the latest details about [SAP BTP, Cloud Foundry Environment quotas and limitations](https://help.sap.com/viewer/3504ec5ef16548778610c7e89cc0eac3/Cloud/en-US/9c7092c7b7ae4d49bc8ae35fdd0e0b18.html#loio9809fa4f02cb4696baea5c23d6eaac94)
- If even after achieving the maximum vertical scale of your instance, your app doesn't reach the desired performance level, try to optimize it.
  Consider scaling it horizontally or different custom solutions to boost performance where it lags.

## Request via Cloud Connector fails with HTTP 407

:::info **Symptoms**
In your application log you find the HTTP error code 407 and the request targets an on-premise system via the SAP Cloud Connector.
:::

**Possible causes:**

> - You cache the `HttpDestination` instance in your application code instead of retrieving it for each request.

Reviewer: ```suggestion
- You are caching the `HttpDestination` instance in your application code instead of retrieving it for each request.
```

>   Since the `HttpDestination` object stores the JWT needed for passing the SAP Cloud Connector, the JWT does not get refreshed after its expiration.

Reviewer: ```suggestion
  Since the `HttpDestination` object stores the JWT needed in the SAP Cloud Connector, the JWT does not get refreshed after its expiration.
```

Author: @Johannes-Schneider Not sure, the JWT is not really needed "in" in the Cloud Connector. It's needed in the request header to pass the Cloud Connector as proxy.

Reviewer: @Emdee89 So in your initial description you meant "passing the SAP Cloud Connector" in the sense of "going through the SAP Cloud Connector", right?
Then I got your intention wrong. I was thinking you meant something like "the JWT is needed to be passed to the SAP Cloud Connector" (and therefore, I assumed the Cloud Connector actually needs the JWT for something).
